### ISSUE TITLE
Getting the user from context doesn't work out of the box.

### ISSUE THREAD

**Issue by KubqoA** [nuwave/lighthouse#119]
Hi, during the video walkthrough I got the authentication part and to injecting the logged in user's id out of context. However, this didn't work for me. I created a simple api route to get the current logged in user using the same way as the default context defined in your GraphQL controller does. I found out that when using the api endpoint with the `auth:api` middleware it worked fine. So it got me thinking whether the endpoint of graphql server uses this middleware and it doesn't. I added it using the lighthouse.php config file like this:
``
'route' => [
   'middleware' => 'auth:api',
],
``
After adding this it worked. However, my question is whether this is the right approach, or if I have missed something. Thanks :)

**Comment by olivernybroe** [comment]
Pretty sure it should be `@inject(context: "user.id" name: "user_id")`.

**Comment by KubqoA** [comment]
Okay, so I discovered what my problem was. The issue is that I didn't change the default auth guard to api, thus my authorization didn't work. When I switched the default auth guard to api everything worked liked charm.

However, I didn't want to change the default auth driver. So I looked for some help and this is my solution if anyone is interested. Firstly I added these lines to my ``lighthouse.php`` config file to indicate that the graphql route should use the api middleware:
```
'route' => [
    'middleware' => 'api',
],
```
Then I created a new middleware I named it UseApiAuthGuard:
```
<?php

namespace App\Http\Middleware;

use Closure;

class UseApiAuthGuard
{
    /**
     * Handle an incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  \Closure  $next
     * @return mixed
     */
    public function handle($request, Closure $next)
    {
        \Auth::shouldUse('api');

        return $next($request);
    }
}
```
I added this middleware to the api middleware group in the Kernel.php and that's it. It forces routes in which have the api middleware to use the api auth guard not the web one.
